ethers-solc/src/compile/output.rs

```diff
@@ -0,0 +1,303 @@
+    /// let mut output = project.compile().unwrap().output();
+    /// let contract = output.remove("Greeter").unwrap();
+    /// # }
+    /// ```
+    pub fn remove(&mut self, contract: impl AsRef<str>) -> Option<Contract> {
+        self.contracts.remove(contract)
+    }
+
+    /// Iterate over all contracts and their names
+    pub fn contracts_iter(&self) -> impl Iterator<Item = (&String, &Contract)> {
+        self.contracts.contracts()
+    }
+
+    /// Iterate over all contracts and their names
+    pub fn contracts_into_iter(self) -> impl Iterator<Item = (String, Contract)> {
+        self.contracts.into_contracts()
+    }
+
+    /// Given the contract file's path and the contract's name, tries to return the contract's
+    /// bytecode, runtime bytecode, and abi
+    pub fn get(&self, path: &str, contract: &str) -> Option<CompactContractRef> {
+        self.contracts.get(path, contract)
+    }
+
+    /// Returns the output's source files and contracts separately, wrapped in helper types that
+    /// provide several helper methods
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use ethers_solc::Project;
+    /// # fn demo(project: Project) {
+    /// let output = project.compile().unwrap().output();
+    /// let (sources, contracts) = output.split();
+    /// # }
+    /// ```
+    pub fn split(self) -> (SourceFiles, VersionedContracts) {
+        (SourceFiles(self.sources), self.contracts)
+    }
+}
+
+/// Helper type to implement display for solc errors
+#[derive(Clone, Debug)]
+pub struct OutputDiagnostics<'a> {
+    compiler_output: &'a AggregatedCompilerOutput,
+    ignored_error_codes: &'a [u64],
+}
+
+impl<'a> OutputDiagnostics<'a> {
+    /// Returns true if there is at least one error of high severity
+    pub fn has_error(&self) -> bool {
+        self.compiler_output.has_error()
+    }
+
+    /// Returns true if there is at least one warning
+    pub fn has_warning(&self) -> bool {
+        self.compiler_output.has_warning(self.ignored_error_codes)
+    }
+
+    /// Returns true if the contract is a expected to be a test
+    fn is_test<T: AsRef<str>>(&self, contract_path: T) -> bool {
+        if contract_path.as_ref().ends_with(".t.sol") {
+            return true
+        }
+
+        self.compiler_output.find(&contract_path).map_or(false, |contract| {
+            contract.abi.map_or(false, |abi| abi.functions.contains_key("IS_TEST"))
+        })
+    }
+}
+
+impl<'a> fmt::Display for OutputDiagnostics<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.has_error() {
+            f.write_str("Compiler run failed")?;
+        } else if self.has_warning() {
+            f.write_str("Compiler run successful (with warnings)")?;
+        } else {
+            f.write_str("Compiler run successful")?;
+        }
+        for err in &self.compiler_output.errors {
+            if err.severity.is_warning() {
+                let is_ignored = err.error_code.as_ref().map_or(false, |code| {
```

Comment:
Hmm this warning code looks correct as well (trying to find where the @mds1 warnings issue is)

Reply:
Let me know if I should patch foundry with this ethers-rs branch and help test compiler warnings being surfaced. I'm lacking some context here—not sure if this is new code, or it just got moved around so it shows up in the PR as a change

Reply:
Yeah once @mattsse thinks this is good for testing, it'd really help to do that. We should also add a test for this